### PR TITLE
[fastapi] add guide for fastapi

### PR DESCRIPTION
**Pull request by cmacht**

### CODE REVIEW

File: source/guide_fastapi.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+Configuration
+=============
+
+Incoming https requests are routed through :manual:`nginx <background-http-stack>` to a :manual:`web backend <web-backends>` where our application server is listening.
+Let's get this working:
+
+Gunicorn
+--------
+First, we are telling gunicorn to spin up several uvicorn processes and listen on port 8000 for incoming requests.
```
Comment (luto) [inline source/guide_fastapi.rst:108]: Where should the content of the code block go? `~/fastapi/conf.py`?
```suggestion
First, we are telling gunicorn to spin up several uvicorn processes and listen on port 8000 for incoming requests. Create the config file ``~/fastapi/conf.py``:
```

File: source/guide_fastapi.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  chdir = app_path
+  workers = 4
+  worker_class = 'uvicorn.workers.UvicornWorker'
+  errorlog =  app_path + '/errors.log'
+
+
+
+This is a minimal working example based on gunicorn's `example config`_, you can find a comprehensive `list of options here`_.
+
+.. note:: Test if everything is working with ``~/fastapi/.env/bin/gunicorn --config ~/fastapi/conf.py``
```
Comment (luto) [inline source/guide_fastapi.rst:127]: This command just sits there and (presumably) waits for connections. There is no output, though. So how do I know that everything is working? Adding a `--check-config` at the end makes gunicorn exit after checking the config.
```suggestion
.. note:: Test if everything is working with ``~/fastapi/.env/bin/gunicorn --config ~/fastapi/conf.py --check-config``. If there is no output, you're good.
```
What do you think?

File: source/guide_fastapi.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+
+Supervisord
+-----------
+Next, create a configuration for supervisord in ``~/etc/services.d/fastapi.ini``:
+
+.. code-block::
+
+  directory=%(ENV_HOME)s/fastapi
+  command=%(ENV_HOME)s/fastapi/.env/bin/gunicorn --config %(ENV_HOME)s/fastapi/conf.py
```
Comment (luto) [inline source/guide_fastapi.rst:137]:
```
[phpbb1@antila fastapi]$ supervisorctl reread
ERROR: CANT_REREAD: File contains no section headers.
file: /home/phpbb1/etc/services.d/fastapi.ini, line: 1
'directory=%(ENV_HOME)s/fastapi\n'
```
This seems to miss the `[program:]` line. How about this?
```suggestion
  [program:fastapi]
  directory=%(ENV_HOME)s/fastapi
  command=%(ENV_HOME)s/fastapi/.env/bin/gunicorn --config %(ENV_HOME)s/fastapi/conf.py
```

**Review by luto — CHANGES_REQUESTED** [review]
Thank you! :tada: I added a few comments to make the guide easier to understand. Also, one fix. Let me know what you think!

File: source/guide_fastapi.rst (hunk 3 above)
Comment (cmacht) [reply source/guide_fastapi.rst:137]: Yes. Good catch!

File: source/guide_fastapi.rst (hunk 1 above)
Comment (cmacht) [reply source/guide_fastapi.rst:108]: Yes, oversight on my part!

File: source/guide_fastapi.rst (hunk 2 above)
Comment (cmacht) [reply source/guide_fastapi.rst:127]: Yap, that's better!

**Comment by cmacht** [comment]
Thanks for the relevant improvements! I applied all suggestions.

**Comment by luto** [comment]
Thank you! :tada:

**Pull request merged**